lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
That's not about splitting federation logic from models. Following to this logic `Retraction::Dispatchable` is actually something like model. So we kinda always use models for federation entity generation & dispatch. Maybe I just picked wrong naming and `Retraction::Dispatchable` should be called `Retraction` and what is called `Retraction` in my diff should be called `RetractionDispatcher`. For usual models the part which does the job submission is in `Diaspora::Federation::Dispatcher` and if we follow that design for retractions then we should probably make a separate `RetractionDispatcher`. Other models don't do their job submission/dispatching themselves, they use dispatchers for this. It makes more sense this way now, and it means that I can try to make another diff for that :) Maybe you will like it more.

But do you think it worth to continue discussion within this PR? It feels somewhat out-of-scope. On the other hand. We're at it now, and we refactor the retractions code in this PR anyway so maybe it isn't bad time to make some more changes.